### PR TITLE
Editorial for Moorio Kart (USACO)

### PR DESCRIPTION
**Pull request by eysbutno**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/gold/usaco-925.mdx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+title: Mooriokart
+author: Justin Ji
+---
+
+[Official Editorial (C++)](https://usaco.org/current/data/sol_mooriokart_platinum_feb19.html)
+
+<Spoiler title = "Hint 1">
+
+The graph is a forest, and we are only allowed to pick one path from each tree
+to reach our total. How can we write a knapsack DP to find our answer?
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-925.mdx:13]: i mean is this not just restating the problem

File: solutions/gold/usaco-925.mdx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+				auto &[sum, freq] = occ[min(y, d + w)];
+				sum = (sum + d + w) % MOD;
+				freq++;
+			}
+			dfs2(v, u, d + w, s);
+		}
+	};
+
+	vector<array<ll, 2>> dp(y + 1);
+	// dp[path_len] = {len_sum, num_paths}
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-925.mdx:95]: why do you put the comment below the thing

File: solutions/gold/usaco-925.mdx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+			}
+		}
+
+		comp.clear();
+		occ.clear();
+		dp = move(dp_t);
+	}
+
+	for (int i = 1; i < k; i++) { dp[y][0] = (dp[y][0] * 2 * i) % MOD; }
+	cout << dp[y][0] << "\n";
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-925.mdx:120]: use endl and my life is yours

File: solutions/gold/usaco-925.mdx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+	vector<array<ll, 2>> dp(y + 1);
+	// dp[path_len] = {len_sum, num_paths}
+	int k = n - m;
+	dp[min(y, k * x)] = {k * x, 1};
+	for (int i = 0; i < n; i++) {
+		if (vis[i]) { continue; }
+		dfs1(i, -1);
+		for (int v : comp) { dfs2(v, -1, 0, v); }
+
+		vector<array<ll, 2>> dp_t(y + 1);
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-925.mdx:103]: what a great name

File: solutions/gold/usaco-925.mdx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+	for (int i = 0; i < m; i++) {
+		int u, v, d;
+		cin >> u >> v >> d;
+		adj[--u].push_back({--v, d});
+		adj[v].push_back({u, d});
+	}
+
+	vector<bool> vis(n);
+	vector<int> comp;
+	function<void(int, int)> dfs1 = [&](int u, int p) {
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-925.mdx:72]: this one too

File: solutions/gold/usaco-925.mdx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+		vis[u] = true;
+		comp.push_back(u);
+		for (auto [v, w] : adj[u]) {
+			if (v == p) { continue; }
+			dfs1(v, u);
+		}
+	};
+
+	map<int, array<int, 2>> occ;
+	function<void(int, int, int, int)> dfs2 = [&](int u, int p, int d, int s) {
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-925.mdx:82]: same here

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/gold/usaco-925.mdx (hunk 6 above)
Comment (TheGamingMousse) [reply solutions/gold/usaco-925.mdx:82]: oops i forgot to make the variable names better, will do tmrw

File: solutions/gold/usaco-925.mdx (hunk 1 above)
Comment (TheGamingMousse) [reply solutions/gold/usaco-925.mdx:13]: true, i mainly added it to lean into hint 2, but I might end up removing it later

File: solutions/gold/usaco-925.mdx [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+
+constexpr int MOD = 1e9 + 7;
```
Comment (ryanchou-dev) [inline solutions/gold/usaco-925.mdx:56]: does this need to be initialized at compile time?

File: solutions/gold/usaco-925.mdx (hunk 1 above)
Comment (ryanchou-dev) [reply solutions/gold/usaco-925.mdx:13]: agree - just combine the two hints into one

**Review by ryanchou-dev — COMMENTED** [review]
just minor things for now, will re-review after kevin's changes are addressed

File: solutions/gold/usaco-925.mdx (hunk 7 above)
Comment (TheGamingMousse) [reply solutions/gold/usaco-925.mdx:56]: i just write this out of habit lol, I'll change it to `const` later

File: solutions/gold/usaco-925.mdx (hunk 2 above)
Comment (TheGamingMousse) [reply solutions/gold/usaco-925.mdx:95]: oops - we're supposed to put it above the thing right? at least I think that's what u do lol

File: solutions/gold/usaco-925.mdx (hunk 1 above)
Comment (TheGamingMousse) [reply solutions/gold/usaco-925.mdx]: ok i combined them - lmk if it needs a bit more work

File: solutions/gold/usaco-925.mdx (hunk 3 above)
Comment (TheGamingMousse) [reply solutions/gold/usaco-925.mdx]: done

File: solutions/gold/usaco-925.mdx (hunk 4 above)
Comment (TheGamingMousse) [reply solutions/gold/usaco-925.mdx]: changed `dp_t` to `new_dp`, hope that's a bit better lol

File: solutions/gold/usaco-925.mdx (hunk 6 above)
Comment (TheGamingMousse) [reply solutions/gold/usaco-925.mdx]: aight i (think) the variable and method names should be a bit better, lmk if they still need changes

**Review by SansPapyrus683 — APPROVED** [review]

**Comment by TheGamingMousse** [comment]
> just minor things for now, will re-review after kevin's changes are addressed

can you check it now? 🙏

**Pull request merged**